great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -276,7 +278,10 @@ def _save_project_config(self) -> None:
         """
         self.variables.save_config()
 
-    @abstractmethod
+    @usage_statistics_enabled_method(
+        event_name=UsageStatsEvents.DATA_CONTEXT_SAVE_EXPECTATION_SUITE,
+        args_payload_fn=save_expectation_suite_usage_statistics,
+    )
```

Comment:
The implementation is pretty much the same across the hierarchy so let's remove some impls and the `@abstracmethod` decorator and put the default impl here. Note we're still overriding in `CloudDataContext` due to the `GXCloudIdentifier` creation logic.

Further work can be done here but for now, I think this feels pretty good.